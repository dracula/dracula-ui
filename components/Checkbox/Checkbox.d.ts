import React, { HTMLAttributes } from 'react';
import { ColorMap } from '../../base/colors';
import { MarginMixin, PaddingMixin } from '../../base/spacing';
export declare const checkboxColors: Partial<ColorMap>;
/** Checkbox Props */
export interface CheckboxProps extends HTMLAttributes<HTMLInputElement>, PaddingMixin, MarginMixin {
    /** The Dracula UI theme color for the checkbox. */
    color?: keyof typeof checkboxColors;
    /** The name of the checkbox. Mirrors the name HTML attribute. */
    name?: string;
    /**
   * A disabled checkbox is unusable and un-clickable.
   */
    disabled?: boolean;
}
/**
 * Checkbox is a styled HTML Input of type checkbox.
 *
 * There are no behavior changes applied to the native HTML tag other
 * than light styling done via CSS in order to keep check boxes accessible.
 */
export declare const Checkbox: React.FC<CheckboxProps>;
