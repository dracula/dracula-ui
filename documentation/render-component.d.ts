import React from 'react';
export declare type ComponentExample = {
    html: string;
    docs: string;
    title: string;
    react?: any;
    screenshot?: string;
    svgData?: string;
};
export declare type SnapshotBuilder = {
    comp: () => React.ReactElement;
    docs: string;
    title: string;
};
export declare function snapshot(title: string, comp: () => React.ReactElement, docs?: string): SnapshotBuilder;
export declare function renderSnapshot(title: string, comp: () => React.ReactElement, docs?: string): ComponentExample;
