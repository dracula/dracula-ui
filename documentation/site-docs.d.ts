import { SnapshotBuilder } from './render-component';
import React from 'react';
interface Documentation {
    basic: () => SnapshotBuilder;
    variations: () => Array<SnapshotBuilder> | SnapshotBuilder;
}
export declare function docs<T>(componentClass: React.FunctionComponent<T> | string, documentation: Documentation): void;
export {};
