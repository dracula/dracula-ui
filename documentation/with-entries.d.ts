import { SnapshotBuilder } from './render-component';
export declare function withEntries<T>(map: T, fn: (name: keyof T, value: T[keyof T]) => SnapshotBuilder): SnapshotBuilder[];
