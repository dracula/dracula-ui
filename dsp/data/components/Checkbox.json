{
  "class": "component",
  "type": "page",
  "id": "Checkbox",
  "name": "Checkbox",
  "last_updated": "2021-03-14T00:11:49.844Z",
  "last_updated_by": "System",
  "description": "Checkbox is a styled HTML Input of type checkbox.\n\nThere are no behavior changes applied to the native HTML tag other\nthan light styling done via CSS in order to keep check boxes accessible.",
  "related_entity_ids": [],
  "tags": ["component"],
  "snippets": {
    "trigger": "checkbox",
    "languages": {
      "javascriptreact": "<div>\n  <Checkbox id=\"one\" defaultChecked={true} color=\"purple\" name=\"demo\" />\n  <label htmlFor=\"one\" className=\"drac-text drac-text-white\">\n    Checked\n  </label>\n</div>;\n",
      "html": "<div>\n  <input\n    type=\"checkbox\"\n    id=\"one\"\n    color=\"purple\"\n    name=\"demo\"\n    class=\"drac-checkbox drac-checkbox-purple\"\n    checked=\"\"\n  /><label for=\"one\" class=\"drac-text drac-text-white\">Checked</label>\n</div>\n"
    }
  },
  "ext_com_draculaui_variations": [
    {
      "title": "Usage",
      "html": "<div>\n  <input\n    type=\"checkbox\"\n    id=\"one\"\n    color=\"purple\"\n    name=\"demo\"\n    class=\"drac-checkbox drac-checkbox-purple\"\n    checked=\"\"\n  /><label for=\"one\" class=\"drac-text drac-text-white\">Checked</label>\n</div>\n",
      "react": "<div>\n  <Checkbox id=\"one\" defaultChecked={true} color=\"purple\" name=\"demo\" />\n  <label htmlFor=\"one\" className=\"drac-text drac-text-white\">\n    Checked\n  </label>\n</div>;\n",
      "docs": "",
      "screenshot": "./dsp/assets/Checkbox.png",
      "svgData": "./dsp/assets/svgs/Checkbox.svg"
    },
    {
      "title": "Disabled",
      "html": "<div style=\"display: flex; flex-direction: column; padding: 3rem\">\n  <div style=\"margin-bottom: 10px\">\n    <input\n      type=\"checkbox\"\n      id=\"one\"\n      disabled=\"\"\n      color=\"white\"\n      name=\"demo\"\n      class=\"drac-checkbox drac-checkbox-white\"\n      checked=\"\"\n    /><label for=\"one\" class=\"drac-text drac-text-white\">Checked</label>\n  </div>\n  <div style=\"margin-bottom: 10px\">\n    <input\n      type=\"checkbox\"\n      id=\"two\"\n      disabled=\"\"\n      color=\"white\"\n      name=\"demo\"\n      class=\"drac-checkbox drac-checkbox-white\"\n    /><label for=\"two\" class=\"drac-text drac-text-white\">Unchecked</label>\n  </div>\n</div>\n",
      "react": "<div style={{ display: \"flex\", flexDirection: \"column\", padding: \"3rem\" }}>\n  <div style={{ marginBottom: 10 }}>\n    <Checkbox\n      id=\"one\"\n      disabled={true}\n      defaultChecked={true}\n      color=\"white\"\n      name=\"demo\"\n    />\n    <label htmlFor=\"one\" className=\"drac-text drac-text-white\">\n      Checked\n    </label>\n  </div>\n  <div style={{ marginBottom: 10 }}>\n    <Checkbox id=\"two\" disabled={true} color=\"white\" name=\"demo\" />\n    <label htmlFor=\"two\" className=\"drac-text drac-text-white\">\n      Unchecked\n    </label>\n  </div>\n</div>;\n",
      "docs": "",
      "screenshot": "./dsp/assets/CheckboxDisabled.png",
      "svgData": "./dsp/assets/svgs/CheckboxDisabled.svg"
    },
    {
      "title": "Colors",
      "html": "<div style=\"display: flex; flex-direction: column; padding: 3rem\">\n  <div style=\"margin-bottom: 10px\">\n    <input\n      type=\"checkbox\"\n      id=\"white\"\n      color=\"white\"\n      name=\"demo\"\n      class=\"drac-checkbox drac-checkbox-white\"\n      checked=\"\"\n    /><label for=\"white\" class=\"drac-text drac-text-white\">Checkbox</label>\n  </div>\n  <div style=\"margin-bottom: 10px\">\n    <input\n      type=\"checkbox\"\n      id=\"cyan\"\n      color=\"cyan\"\n      name=\"demo\"\n      class=\"drac-checkbox drac-checkbox-cyan\"\n      checked=\"\"\n    /><label for=\"cyan\" class=\"drac-text drac-text-white\">Checkbox</label>\n  </div>\n  <div style=\"margin-bottom: 10px\">\n    <input\n      type=\"checkbox\"\n      id=\"green\"\n      color=\"green\"\n      name=\"demo\"\n      class=\"drac-checkbox drac-checkbox-green\"\n      checked=\"\"\n    /><label for=\"green\" class=\"drac-text drac-text-white\">Checkbox</label>\n  </div>\n  <div style=\"margin-bottom: 10px\">\n    <input\n      type=\"checkbox\"\n      id=\"orange\"\n      color=\"orange\"\n      name=\"demo\"\n      class=\"drac-checkbox drac-checkbox-orange\"\n      checked=\"\"\n    /><label for=\"orange\" class=\"drac-text drac-text-white\">Checkbox</label>\n  </div>\n  <div style=\"margin-bottom: 10px\">\n    <input\n      type=\"checkbox\"\n      id=\"pink\"\n      color=\"pink\"\n      name=\"demo\"\n      class=\"drac-checkbox drac-checkbox-pink\"\n      checked=\"\"\n    /><label for=\"pink\" class=\"drac-text drac-text-white\">Checkbox</label>\n  </div>\n  <div style=\"margin-bottom: 10px\">\n    <input\n      type=\"checkbox\"\n      id=\"purple\"\n      color=\"purple\"\n      name=\"demo\"\n      class=\"drac-checkbox drac-checkbox-purple\"\n      checked=\"\"\n    /><label for=\"purple\" class=\"drac-text drac-text-white\">Checkbox</label>\n  </div>\n  <div style=\"margin-bottom: 10px\">\n    <input\n      type=\"checkbox\"\n      id=\"red\"\n      color=\"red\"\n      name=\"demo\"\n      class=\"drac-checkbox drac-checkbox-red\"\n      checked=\"\"\n    /><label for=\"red\" class=\"drac-text drac-text-white\">Checkbox</label>\n  </div>\n  <div style=\"margin-bottom: 10px\">\n    <input\n      type=\"checkbox\"\n      id=\"yellow\"\n      color=\"yellow\"\n      name=\"demo\"\n      class=\"drac-checkbox drac-checkbox-yellow\"\n      checked=\"\"\n    /><label for=\"yellow\" class=\"drac-text drac-text-white\">Checkbox</label>\n  </div>\n  <div style=\"margin-bottom: 10px\">\n    <input\n      type=\"checkbox\"\n      id=\"animated\"\n      color=\"animated\"\n      name=\"demo\"\n      class=\"drac-checkbox\"\n      checked=\"\"\n    /><label for=\"animated\" class=\"drac-text drac-text-white\">Checkbox</label>\n  </div>\n  <div style=\"margin-bottom: 10px\">\n    <input\n      type=\"checkbox\"\n      id=\"cyanGreen\"\n      color=\"cyanGreen\"\n      name=\"demo\"\n      class=\"drac-checkbox\"\n      checked=\"\"\n    /><label for=\"cyanGreen\" class=\"drac-text drac-text-white\">Checkbox</label>\n  </div>\n  <div style=\"margin-bottom: 10px\">\n    <input\n      type=\"checkbox\"\n      id=\"pinkPurple\"\n      color=\"pinkPurple\"\n      name=\"demo\"\n      class=\"drac-checkbox\"\n      checked=\"\"\n    /><label for=\"pinkPurple\" class=\"drac-text drac-text-white\">Checkbox</label>\n  </div>\n  <div style=\"margin-bottom: 10px\">\n    <input\n      type=\"checkbox\"\n      id=\"purpleCyan\"\n      color=\"purpleCyan\"\n      name=\"demo\"\n      class=\"drac-checkbox\"\n      checked=\"\"\n    /><label for=\"purpleCyan\" class=\"drac-text drac-text-white\">Checkbox</label>\n  </div>\n  <div style=\"margin-bottom: 10px\">\n    <input\n      type=\"checkbox\"\n      id=\"yellowPink\"\n      color=\"yellowPink\"\n      name=\"demo\"\n      class=\"drac-checkbox\"\n      checked=\"\"\n    /><label for=\"yellowPink\" class=\"drac-text drac-text-white\">Checkbox</label>\n  </div>\n</div>\n",
      "react": "<div style={{ display: \"flex\", flexDirection: \"column\", padding: \"3rem\" }}>\n  <div style={{ marginBottom: 10 }}>\n    <Checkbox id=\"white\" defaultChecked={true} color=\"white\" name=\"demo\" />\n    <label htmlFor=\"white\" className=\"drac-text drac-text-white\">\n      Checkbox\n    </label>\n  </div>\n  <div style={{ marginBottom: 10 }}>\n    <Checkbox id=\"cyan\" defaultChecked={true} color=\"cyan\" name=\"demo\" />\n    <label htmlFor=\"cyan\" className=\"drac-text drac-text-white\">\n      Checkbox\n    </label>\n  </div>\n  <div style={{ marginBottom: 10 }}>\n    <Checkbox id=\"green\" defaultChecked={true} color=\"green\" name=\"demo\" />\n    <label htmlFor=\"green\" className=\"drac-text drac-text-white\">\n      Checkbox\n    </label>\n  </div>\n  <div style={{ marginBottom: 10 }}>\n    <Checkbox id=\"orange\" defaultChecked={true} color=\"orange\" name=\"demo\" />\n    <label htmlFor=\"orange\" className=\"drac-text drac-text-white\">\n      Checkbox\n    </label>\n  </div>\n  <div style={{ marginBottom: 10 }}>\n    <Checkbox id=\"pink\" defaultChecked={true} color=\"pink\" name=\"demo\" />\n    <label htmlFor=\"pink\" className=\"drac-text drac-text-white\">\n      Checkbox\n    </label>\n  </div>\n  <div style={{ marginBottom: 10 }}>\n    <Checkbox id=\"purple\" defaultChecked={true} color=\"purple\" name=\"demo\" />\n    <label htmlFor=\"purple\" className=\"drac-text drac-text-white\">\n      Checkbox\n    </label>\n  </div>\n  <div style={{ marginBottom: 10 }}>\n    <Checkbox id=\"red\" defaultChecked={true} color=\"red\" name=\"demo\" />\n    <label htmlFor=\"red\" className=\"drac-text drac-text-white\">\n      Checkbox\n    </label>\n  </div>\n  <div style={{ marginBottom: 10 }}>\n    <Checkbox id=\"yellow\" defaultChecked={true} color=\"yellow\" name=\"demo\" />\n    <label htmlFor=\"yellow\" className=\"drac-text drac-text-white\">\n      Checkbox\n    </label>\n  </div>\n  <div style={{ marginBottom: 10 }}>\n    <Checkbox\n      id=\"animated\"\n      defaultChecked={true}\n      color=\"animated\"\n      name=\"demo\"\n    />\n    <label htmlFor=\"animated\" className=\"drac-text drac-text-white\">\n      Checkbox\n    </label>\n  </div>\n  <div style={{ marginBottom: 10 }}>\n    <Checkbox\n      id=\"cyanGreen\"\n      defaultChecked={true}\n      color=\"cyanGreen\"\n      name=\"demo\"\n    />\n    <label htmlFor=\"cyanGreen\" className=\"drac-text drac-text-white\">\n      Checkbox\n    </label>\n  </div>\n  <div style={{ marginBottom: 10 }}>\n    <Checkbox\n      id=\"pinkPurple\"\n      defaultChecked={true}\n      color=\"pinkPurple\"\n      name=\"demo\"\n    />\n    <label htmlFor=\"pinkPurple\" className=\"drac-text drac-text-white\">\n      Checkbox\n    </label>\n  </div>\n  <div style={{ marginBottom: 10 }}>\n    <Checkbox\n      id=\"purpleCyan\"\n      defaultChecked={true}\n      color=\"purpleCyan\"\n      name=\"demo\"\n    />\n    <label htmlFor=\"purpleCyan\" className=\"drac-text drac-text-white\">\n      Checkbox\n    </label>\n  </div>\n  <div style={{ marginBottom: 10 }}>\n    <Checkbox\n      id=\"yellowPink\"\n      defaultChecked={true}\n      color=\"yellowPink\"\n      name=\"demo\"\n    />\n    <label htmlFor=\"yellowPink\" className=\"drac-text drac-text-white\">\n      Checkbox\n    </label>\n  </div>\n</div>;\n",
      "docs": "",
      "screenshot": "./dsp/assets/CheckboxColors.png",
      "svgData": "./dsp/assets/svgs/CheckboxColors.svg"
    }
  ],
  "ext_com_draculaui_props": {
    "color": {
      "defaultValue": null,
      "description": "The Dracula UI theme color for the checkbox.",
      "name": "color",
      "parent": {
        "fileName": "src/components/Checkbox/Checkbox.tsx",
        "name": "CheckboxProps"
      },
      "declarations": [
        {
          "fileName": "src/components/Checkbox/Checkbox.tsx",
          "name": "CheckboxProps"
        }
      ],
      "required": false,
      "type": {
        "name": "\"white\" | \"cyan\" | \"green\" | \"orange\" | \"pink\" | \"purple\" | \"red\" | \"yellow\" | \"purpleCyan\" | \"yellowPink\" | \"cyanGreen\" | \"pinkPurple\" | \"animated\" | undefined"
      }
    },
    "name": {
      "defaultValue": null,
      "description": "The name of the checkbox. Mirrors the name HTML attribute.",
      "name": "name",
      "parent": {
        "fileName": "src/components/Checkbox/Checkbox.tsx",
        "name": "CheckboxProps"
      },
      "declarations": [
        {
          "fileName": "src/components/Checkbox/Checkbox.tsx",
          "name": "CheckboxProps"
        }
      ],
      "required": false,
      "type": { "name": "string | undefined" }
    },
    "disabled": {
      "defaultValue": null,
      "description": "",
      "name": "disabled",
      "parent": {
        "fileName": "src/components/Checkbox/Checkbox.tsx",
        "name": "CheckboxProps"
      },
      "declarations": [
        {
          "fileName": "src/components/Checkbox/Checkbox.tsx",
          "name": "CheckboxProps"
        }
      ],
      "required": false,
      "type": { "name": "boolean | undefined" }
    }
  },
  "ext_com_draculaui_docgen": {
    "tags": {},
    "description": "Checkbox is a styled HTML Input of type checkbox.\n\nThere are no behavior changes applied to the native HTML tag other\nthan light styling done via CSS in order to keep check boxes accessible.",
    "displayName": "Checkbox",
    "methods": [],
    "props": {
      "color": {
        "defaultValue": null,
        "description": "The Dracula UI theme color for the checkbox.",
        "name": "color",
        "parent": {
          "fileName": "src/components/Checkbox/Checkbox.tsx",
          "name": "CheckboxProps"
        },
        "declarations": [
          {
            "fileName": "src/components/Checkbox/Checkbox.tsx",
            "name": "CheckboxProps"
          }
        ],
        "required": false,
        "type": {
          "name": "\"white\" | \"cyan\" | \"green\" | \"orange\" | \"pink\" | \"purple\" | \"red\" | \"yellow\" | \"purpleCyan\" | \"yellowPink\" | \"cyanGreen\" | \"pinkPurple\" | \"animated\" | undefined"
        }
      },
      "name": {
        "defaultValue": null,
        "description": "The name of the checkbox. Mirrors the name HTML attribute.",
        "name": "name",
        "parent": {
          "fileName": "src/components/Checkbox/Checkbox.tsx",
          "name": "CheckboxProps"
        },
        "declarations": [
          {
            "fileName": "src/components/Checkbox/Checkbox.tsx",
            "name": "CheckboxProps"
          }
        ],
        "required": false,
        "type": { "name": "string | undefined" }
      },
      "disabled": {
        "defaultValue": null,
        "description": "",
        "name": "disabled",
        "parent": {
          "fileName": "src/components/Checkbox/Checkbox.tsx",
          "name": "CheckboxProps"
        },
        "declarations": [
          {
            "fileName": "src/components/Checkbox/Checkbox.tsx",
            "name": "CheckboxProps"
          }
        ],
        "required": false,
        "type": { "name": "boolean | undefined" }
      }
    }
  }
}
