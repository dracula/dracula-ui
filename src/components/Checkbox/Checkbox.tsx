import React, { HTMLAttributes } from 'react'
import cx from 'classnames/dedupe'
import { ColorMap } from '../../base/colors'
import {
  cleanProps,
  marginMixin,
  MarginMixin,
  paddingMixin,
  PaddingMixin
} from '../../base/spacing'

export const checkboxColors: Partial<ColorMap> = {
  white: 'drac-checkbox-white',
  cyan: 'drac-checkbox-cyan',
  green: 'drac-checkbox-green',
  orange: 'drac-checkbox-orange',
  pink: 'drac-checkbox-pink',
  purple: 'drac-checkbox-purple',
  red: 'drac-checkbox-red',
  yellow: 'drac-checkbox-yellow'
}

/** Checkbox Props */
export interface CheckboxProps
  extends HTMLAttributes<HTMLInputElement>,
    PaddingMixin,
    MarginMixin {
  /** The Dracula UI theme color for the checkbox. */
  color?: keyof typeof checkboxColors
  /** The name of the checkbox. Mirrors the name HTML attribute. */
  name?: string

  /**
   * A disabled checkbox is unusable and un-clickable.
   */
  disabled?: boolean
}

/**
 * Checkbox is a styled HTML Input of type checkbox.
 *
 * There are no behavior changes applied to the native HTML tag other
 * than light styling done via CSS in order to keep check boxes accessible.
 */
export const Checkbox: React.FC<CheckboxProps> = (props: CheckboxProps) => {
  const { className, color, name, disabled, ...htmlProps } = props

  const finalProps = {
    name,
    disabled,
    ...htmlProps,
    className: cx(
      `drac-checkbox`,
      className,
      color && checkboxColors[color],
      ...paddingMixin(props),
      ...marginMixin(props)
    )
  }

  return <input type="checkbox" {...cleanProps(finalProps)} />
}

Checkbox.displayName = 'Checkbox'
