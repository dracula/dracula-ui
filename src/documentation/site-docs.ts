import {
  ComponentExample,
  renderSnapshot,
  SnapshotBuilder
} from './render-component'
import React from 'react'
import fs from 'fs-extra'
import { pretty } from './pretty'
import { getDocGen } from '../documentation/docgen/doc-generator'
import { toDSP } from '../documentation/dsp/component-generator'
import puppeteer from 'puppeteer'
import { componentScreenshot } from './screenshots/component-screenshot'
import os from 'os'

interface Documentation {
  basic: () => SnapshotBuilder
  variations: () => Array<SnapshotBuilder> | SnapshotBuilder
}

jest.setTimeout(10000)

const RECORDING = process.env.RECORDING

export function docs<T>(
  componentClass: React.FunctionComponent<T> | string,
  documentation: Documentation
) {
  const name =
    typeof componentClass === 'string'
      ? componentClass
      : componentClass.displayName ?? componentClass.name

  const dspPath = `${process.cwd()}/dsp`

  describe(`Site: ${name}`, () => {
    let browser: puppeteer.Browser

    beforeAll(async () => {
      if (!RECORDING) {
        return
      }

      let executablePath = undefined
      if (os.arch() === 'arm64') {
        executablePath = '/Applications/Chromium.app/Contents/MacOS/Chromium'
      }

      browser = await puppeteer.launch({
        headless: true,
        devtools: true,
        executablePath
      })
    })

    let examples: Record<string, ComponentExample> = {}

    afterAll(async () => {
      if (!RECORDING) {
        return
      }

      await browser.close()
      const docGen = getDocGen(name)
      const dsp = toDSP(name, examples, docGen)

      await fs.mkdirp(`${dspPath}/data/components/`).catch(() => {
        // folder already exists
      })

      const path = `${dspPath}/data/components/${name}.json`
      await fs.writeFile(path, pretty(JSON.stringify(dsp), 'json'))
    })

    test('Basic Usage', async () => {
      const variation = documentation.basic()
      const snapshot = renderSnapshot(
        variation.title,
        variation.comp,
        variation.docs
      )
      expect(snapshot).toMatchSnapshot()

      if (RECORDING) {
        const [screenshot, svg] = await componentScreenshot(
          browser,
          snapshot,
          null,
          name
        )

        snapshot.screenshot = screenshot
        snapshot.svgData = svg
      }

      examples = {
        ...examples,
        basic: snapshot
      }
    })

    const variations: SnapshotBuilder[] = ([] as SnapshotBuilder[]).concat(
      documentation.variations()
    )

    variations.forEach((variation) => {
      test(variation.title, async () => {
        const snapshot = renderSnapshot(
          variation.title,
          variation.comp,
          variation.docs
        )
        expect(snapshot).toMatchSnapshot()

        if (RECORDING) {
          const [screenshot, svg] = await componentScreenshot(
            browser,
            snapshot,
            variation,
            name
          )

          snapshot.screenshot = screenshot
          snapshot.svgData = svg
        }

        examples = {
          ...examples,
          [variation.title]: snapshot
        }
      })
    })
  })
}
