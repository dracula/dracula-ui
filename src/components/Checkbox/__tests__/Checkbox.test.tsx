import { Checkbox } from '@/components/Checkbox/Checkbox'
import {
  buildSnapshot,
  SnapshotBuilder
} from '@/story-helpers/render-component'
import { siteDocs } from '@/documentation/site-docs'
import {
  CheckboxNormal,
  CheckboxDisabled,
  CheckboxColors
} from '@/components/Checkbox/__stories__/Checkbox.stories'

siteDocs(Checkbox, {
  basicUsage() {
    return buildSnapshot('Usage', CheckboxNormal)
  },
  variations() {
    return [
      buildSnapshot('Disabled', CheckboxDisabled),
      buildSnapshot('Colors', CheckboxColors)
    ] as SnapshotBuilder[]
  }
})
