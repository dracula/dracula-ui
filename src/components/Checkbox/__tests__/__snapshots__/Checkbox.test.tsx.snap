// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Site: Checkbox Basic Usage 1`] = `
Object {
  "docs": "Checkbox allows for styling HTML checkboxes with Dracula UI styles.",
  "html": "<div>
  <input
    type=\\"checkbox\\"
    name=\\"demo\\"
    id=\\"one\\"
    class=\\"drac-checkbox drac-checkbox-purple\\"
    checked=\\"\\"
  /><label for=\\"one\\" class=\\"drac-text drac-text-white\\">Checked</label>
</div>
",
  "react": "<div>
  <Checkbox id=\\"one\\" defaultChecked={true} color=\\"purple\\" name=\\"demo\\" />
  <label htmlFor=\\"one\\" className=\\"drac-text drac-text-white\\">
    Checked
  </label>
</div>;
",
  "title": "Usage",
}
`;

exports[`Site: Checkbox Colors 1`] = `
Object {
  "docs": "Checkboxes can be customized to use any of the Dracula UI theme colors, including all solid, gradient, and animated colors.",
  "html": "<div style=\\"display: flex; flex-direction: column\\">
  <div style=\\"margin-bottom: 10px\\">
    <input
      type=\\"checkbox\\"
      name=\\"demo\\"
      id=\\"white\\"
      class=\\"drac-checkbox drac-checkbox-white\\"
      checked=\\"\\"
    /><label for=\\"white\\" class=\\"drac-text drac-text-white\\">Checkbox</label>
  </div>
  <div style=\\"margin-bottom: 10px\\">
    <input
      type=\\"checkbox\\"
      name=\\"demo\\"
      id=\\"cyan\\"
      class=\\"drac-checkbox drac-checkbox-cyan\\"
      checked=\\"\\"
    /><label for=\\"cyan\\" class=\\"drac-text drac-text-white\\">Checkbox</label>
  </div>
  <div style=\\"margin-bottom: 10px\\">
    <input
      type=\\"checkbox\\"
      name=\\"demo\\"
      id=\\"green\\"
      class=\\"drac-checkbox drac-checkbox-green\\"
      checked=\\"\\"
    /><label for=\\"green\\" class=\\"drac-text drac-text-white\\">Checkbox</label>
  </div>
  <div style=\\"margin-bottom: 10px\\">
    <input
      type=\\"checkbox\\"
      name=\\"demo\\"
      id=\\"orange\\"
      class=\\"drac-checkbox drac-checkbox-orange\\"
      checked=\\"\\"
    /><label for=\\"orange\\" class=\\"drac-text drac-text-white\\">Checkbox</label>
  </div>
  <div style=\\"margin-bottom: 10px\\">
    <input
      type=\\"checkbox\\"
      name=\\"demo\\"
      id=\\"pink\\"
      class=\\"drac-checkbox drac-checkbox-pink\\"
      checked=\\"\\"
    /><label for=\\"pink\\" class=\\"drac-text drac-text-white\\">Checkbox</label>
  </div>
  <div style=\\"margin-bottom: 10px\\">
    <input
      type=\\"checkbox\\"
      name=\\"demo\\"
      id=\\"purple\\"
      class=\\"drac-checkbox drac-checkbox-purple\\"
      checked=\\"\\"
    /><label for=\\"purple\\" class=\\"drac-text drac-text-white\\">Checkbox</label>
  </div>
  <div style=\\"margin-bottom: 10px\\">
    <input
      type=\\"checkbox\\"
      name=\\"demo\\"
      id=\\"red\\"
      class=\\"drac-checkbox drac-checkbox-red\\"
      checked=\\"\\"
    /><label for=\\"red\\" class=\\"drac-text drac-text-white\\">Checkbox</label>
  </div>
  <div style=\\"margin-bottom: 10px\\">
    <input
      type=\\"checkbox\\"
      name=\\"demo\\"
      id=\\"yellow\\"
      class=\\"drac-checkbox drac-checkbox-yellow\\"
      checked=\\"\\"
    /><label for=\\"yellow\\" class=\\"drac-text drac-text-white\\">Checkbox</label>
  </div>
</div>
",
  "react": "<div style={{ display: \\"flex\\", flexDirection: \\"column\\" }}>
  <div style={{ marginBottom: 10 }}>
    <Checkbox id=\\"white\\" defaultChecked={true} color=\\"white\\" name=\\"demo\\" />
    <label htmlFor=\\"white\\" className=\\"drac-text drac-text-white\\">
      Checkbox
    </label>
  </div>
  <div style={{ marginBottom: 10 }}>
    <Checkbox id=\\"cyan\\" defaultChecked={true} color=\\"cyan\\" name=\\"demo\\" />
    <label htmlFor=\\"cyan\\" className=\\"drac-text drac-text-white\\">
      Checkbox
    </label>
  </div>
  <div style={{ marginBottom: 10 }}>
    <Checkbox id=\\"green\\" defaultChecked={true} color=\\"green\\" name=\\"demo\\" />
    <label htmlFor=\\"green\\" className=\\"drac-text drac-text-white\\">
      Checkbox
    </label>
  </div>
  <div style={{ marginBottom: 10 }}>
    <Checkbox id=\\"orange\\" defaultChecked={true} color=\\"orange\\" name=\\"demo\\" />
    <label htmlFor=\\"orange\\" className=\\"drac-text drac-text-white\\">
      Checkbox
    </label>
  </div>
  <div style={{ marginBottom: 10 }}>
    <Checkbox id=\\"pink\\" defaultChecked={true} color=\\"pink\\" name=\\"demo\\" />
    <label htmlFor=\\"pink\\" className=\\"drac-text drac-text-white\\">
      Checkbox
    </label>
  </div>
  <div style={{ marginBottom: 10 }}>
    <Checkbox id=\\"purple\\" defaultChecked={true} color=\\"purple\\" name=\\"demo\\" />
    <label htmlFor=\\"purple\\" className=\\"drac-text drac-text-white\\">
      Checkbox
    </label>
  </div>
  <div style={{ marginBottom: 10 }}>
    <Checkbox id=\\"red\\" defaultChecked={true} color=\\"red\\" name=\\"demo\\" />
    <label htmlFor=\\"red\\" className=\\"drac-text drac-text-white\\">
      Checkbox
    </label>
  </div>
  <div style={{ marginBottom: 10 }}>
    <Checkbox id=\\"yellow\\" defaultChecked={true} color=\\"yellow\\" name=\\"demo\\" />
    <label htmlFor=\\"yellow\\" className=\\"drac-text drac-text-white\\">
      Checkbox
    </label>
  </div>
</div>;
",
  "title": "Colors",
}
`;
